## 단일 표본 검정

한 커피 제조회사에서는 새로 출시한 커피의 카페인 함량이 평균 95mg 미만이라고 주장하였다. 그 주장이 사실인가를 알아보기 위하여 25개의 커피 샘플을 무작위(랜덤)로 추출하였다. 커피 제조회사의 주장이 타당한지를 유의수준 5%에서 검정하시오.

- 귀무가설(H0): μ ≥ 95mg
- 대립가설(H1): μ < 95mg
1. 표본 데이터의 평균을 구하시오.
2. Shapiro-Wilk의 p-value를 구하시오.
3. 단일 표본 t-검정의 검정통계량을 구하시오.
4. 단일 표본 t-검정의p-value를 구하시오.
5. 유의 수준 0.05하에서 귀무가설을 기준으로 검정의 결과를 (채택/기각)중 선택하여 입력하시오.

In [ ]:
import pandas as pd
df = pd.DataFrame({
    'Caffeine(mg)': [
        94.2, 93.7, 95.5, 93.9, 94.0, 95.2, 94.7, 93.5, 92.8, 94.4,
        93.8, 94.6, 93.3, 95.1, 94.3, 94.9, 93.9, 94.8, 95.0, 94.2,
        93.7, 94.4, 95.1, 94.0, 93.6
    ]
})

## 독립 표본 검정

스마트폰 배터리의 충전 시간을 줄이기 위한 새로운 충전기를 개발하였다. 개발된 충전기의 효과를 검정하기 위해 스마트폰 사용자 집단을 두 그룹으로 나누어 한 그룹에는 새로운 충전기를 제공하고, 나머지 그룹에는 기존의 충전기를 제공하였다. 그 후, 두 그룹의 평균 충전 완료 시간을 비교하였다. 두 그룹 모두 충전 시간은 정규분포를 따르며 분산은 같다고 가정한다. 새로운 충전기가 더 빠르다고 할 수 있는지를 유의수준 5%하에서 검정하시오. 여기서 두 집단은 모두 정규분포를 따르고 분산은 같다고 가정한다. (μ1: 새로운 충전기 집단의 모평균, μ2: 기존 충전기 집단의 모평균)

- 귀무가설(H0): μ1 ≥ μ2
- 대립가설(H1): μ1 < μ2
1. 위의 가설을 검정하기 위한 검정 통계량을 구하시오.
2. 위의 통계량에 대한 p-value를 구하시오.
3. 유의 수준 0.05하에서 귀무가설을 기준으로 검정의 결과를 (채택/기각)중 선택하여 입력하시오.

In [ ]:
# 데이터
import pandas as pd
df = pd.DataFrame({
    '충전기': ['New'] * 10 + ['Old'] * 10,
    '충전시간': [
        1.5, 1.6, 1.4, 1.7, 1.5, 1.6, 1.7, 1.4, 1.6, 1.5,
        1.7, 1.8, 1.7, 1.9, 1.8, 1.7, 1.8, 1.9, 1.7, 1.6
    ]
})

## 대응 표본 검정

딴짓 연구소에서는 새로 개발한 교육과정에 참여 하면 학습 시간이 단축한다고 주장한다. 10명의 학생들을 대상으로 기존의 교육 방법과 새로운 교육 방법을 적용한 결과에 대하여 유의수준 5%에서 새로운 교육 방법이 교육 시간을 단축시켰는지 검정하시오. (단, 모집단은 정규분포를 가정한다.)

- μd = (새로운 교수 방법 – 기존 교수 방법)의 평균
- 귀무가설(H0): μd = 0
- 대립가설(H1): μd < 0
1. μd의 표본평균을 구하시오.
2. 위의 가설을 검정하기 위한 검정 통계량을 구하시오.
3. 위의 통계량에 대한 p-value를 구하시오.
4. 유의 수준 0.05하에서 귀무가설을 기준으로 검정의 결과를 (채택/기각)중 선택하여 입력하시오.

In [ ]:
import pandas as pd
df = pd.DataFrame({
    'User': list(range(1, 11)),
    '기존방법': [60.4, 60.7, 60.5, 60.3, 60.8, 60.6, 60.2, 60.5, 60.7, 60.4],
    '새로운방법': [59.8, 60.2, 60.1, 59.9, 59.7, 58.4, 57.0, 60.3, 59.6, 59.8]
})

## 단일 표본 검정

In [2]:
# 데이터
import pandas as pd
df = pd.DataFrame({
    'Caffeine(mg)': [
        94.2, 93.7, 95.5, 93.9, 94.0, 95.2, 94.7, 93.5, 92.8, 94.4,
        93.8, 94.6, 93.3, 95.1, 94.3, 94.9, 93.9, 94.8, 95.0, 94.2,
        93.7, 94.4, 95.1, 94.0, 93.6
    ]
})

한 커피 제조회사에서는 새로 출시한 커피의 카페인 함량이 평균 95mg 미만이라고 주장하였다. 그 주장이 사실인가를 알아보기 위하여 25개의 커피 샘플을 무작위(랜덤)로 추출하였다. 커피 제조회사의 주장이 타당한지를 유의수준 5%에서 검정하시오.

- 귀무가설(H0): μ ≥ 95mg
- 대립가설(H1): μ < 95mg
1. 표본 데이터의 평균을 구하시오.
2. Shapiro-Wilk의 p-value를 구하시오.
3. 단일 표본 t-검정의 검정통계량을 구하시오.
4. 단일 표본 t-검정의p-value를 구하시오.
5. 유의 수준 0.05하에서 귀무가설을 기준으로 검정의 결과를 (채택/기각)중 선택하여 입력하시오.

In [3]:
# 1. 표본 평균
print(df.mean())

# 2. 정규성 검정
from scipy import stats
print(stats.shapiro(df['Caffeine(mg)']))

# 3~5.단일 표본 t-검정
print(stats.ttest_1samp(df['Caffeine(mg)'], 95, alternative='less'))

Caffeine(mg)    94.264
dtype: float64
ShapiroResult(statistic=np.float64(0.9826578166170536), pvalue=np.float64(0.9322031137746971))
TtestResult(statistic=np.float64(-5.501737036221897), pvalue=np.float64(5.8686553916715e-06), df=np.int64(24))


In [ ]:
statistic, pvalue = stats.ttest_1samp(df['Caffeine(mg)'], 95, alternative='less')
print("{:.10f}".format(pvalue))

0.0000058687


1. 94.264
2. 0.9321980476379395
3. -5.501737036221897
4. 0.0000058687(5.8686553916715e-06)
5. 기각

## 독립 표본 검정

스마트폰 배터리의 충전 시간을 줄이기 위한 새로운 충전기를 개발하였다. 개발된 충전기의 효과를 검정하기 위해 스마트폰 사용자 집단을 두 그룹으로 나누어 한 그룹에는 새로운 충전기를 제공하고, 나머지 그룹에는 기존의 충전기를 제공하였다. 그 후, 두 그룹의 평균 충전 완료 시간을 비교하였다. 두 그룹 모두 충전 시간은 정규분포를 따르며 분산은 같다고 가정한다. 새로운 충전기가 더 빠르다고 할 수 있는지를 유의수준 5%하에서 검정하시오. 여기서 두 집단은 모두 정규분포를 따르고 분산은 같다고 가정한다. (μ1: 새로운 충전기 집단의 모평균, μ2: 기존 충전기 집단의 모평균)

- 귀무가설(H0): μ1 ≥ μ2
- 대립가설(H1): μ1 < μ2
1. 위의 가설을 검정하기 위한 검정 통계량을 구하시오.
2. 위의 통계량에 대한 p-value를 구하시오.
3. 유의 수준 0.05하에서 귀무가설을 기준으로 검정의 결과를 (채택/기각)중 선택하여 입력하시오.

In [ ]:
# 데이터
import pandas as pd
df = pd.DataFrame({
    '충전기': ['New'] * 10 + ['Old'] * 10,
    '충전시간': [
        1.5, 1.6, 1.4, 1.7, 1.5, 1.6, 1.7, 1.4, 1.6, 1.5,
        1.7, 1.8, 1.7, 1.9, 1.8, 1.7, 1.8, 1.9, 1.7, 1.6
    ]
})
print(df.head(2))

   충전기  충전시간
0  New   1.5
1  New   1.6


In [ ]:
# 1~3. 독립 표본 t-검정
new_cond = df['충전기'] == 'New'
old_cond = df['충전기'] == 'Old'
print(stats.ttest_ind(df[new_cond]['충전시간'], df[old_cond]['충전시간'], alternative='less', equal_var=True))

TtestResult(statistic=np.float64(-4.582575694955849), pvalue=np.float64(0.00011546547787696304), df=np.float64(18.0))


1. -4.582575694955849
2. 0.00011546547787696304
3. 기각


## 대응 표본 검정

딴짓 연구소에서는 새로 개발한 교육과정에 참여 하면 학습 시간이 단축한다고 주장한다. 10명의 학생들을 대상으로 기존의 교육 방법과 새로운 교육 방법을 적용한 결과에 대하여 유의수준 5%에서 새로운 교육 방법이 교육 시간을 단축시켰는지 검정하시오. (단, 모집단은 정규분포를 가정한다.)

- μd = (새로운 교수 방법 – 기존 교수 방법)의 평균
- 귀무가설(H0): μd = 0
- 대립가설(H1): μd < 0
1. μd의 표본평균을 구하시오.
2. 위의 가설을 검정하기 위한 검정 통계량을 구하시오.
3. 위의 통계량에 대한 p-value를 구하시오.
4. 유의 수준 0.05하에서 귀무가설을 기준으로 검정의 결과를 (채택/기각)중 선택하여 입력하시오.

In [ ]:
# 데이터
import pandas as pd
df = pd.DataFrame({
    'User': list(range(1, 11)),
    '기존방법': [60.4, 60.7, 60.5, 60.3, 60.8, 60.6, 60.2, 60.5, 60.7, 60.4],
    '새로운방법': [59.8, 60.2, 60.1, 59.9, 59.7, 58.4, 57.0, 60.3, 59.6, 59.8]
})
print(df.head(2))

   User  기존방법  새로운방법
0     1  60.4   59.8
1     2  60.7   60.2


In [ ]:
# 1. 표본 평균
df['diff'] = df['새로운방법'] - df['기존방법']
print(df['diff'].mean())

# 2~4. 대응 표본 t-검정
print(stats.ttest_rel(df['새로운방법'], df['기존방법'], alternative='less'))

-1.0300000000000005
TtestResult(statistic=np.float64(-3.407973078114844), pvalue=np.float64(0.0038872633380070652), df=np.int64(9))


1. -1.0300000000000005
2. -3.407973078114844
3. 0.0038872633380070652
4. 기각
